pkg/build/netbsd.go

```diff
@@ -66,3 +73,47 @@ func (ctx netbsd) clean(kernelDir string) error {
 	// machine.
 	return nil
 }
+
+// Copy the compiled kernel to the qemu disk image using ssh
+func CopyKernelToDisk(outputDir string) error {
+	temp := []byte(`
+{
+	"qemu": "qemu-system-x86_64",
+	"cpu": 1,
+	"count": 1,
+	"mem": 1024
+}	`)
+	VMconfig := (*json.RawMessage)(&temp)
+	// Create config for booting the disk image
+	cfg := &mgrconfig.Config{
+		Workdir:      outputDir,
+		Image:        filepath.Join(outputDir, "image"),
+		SSHKey:       filepath.Join(outputDir, "key"),
+		SSHUser:      "root",
+		TargetOS:     "netbsd",
+		TargetVMArch: "amd64",
+		Type:         "qemu",
+		VM:           *VMconfig,
+	}
+	// Create a VM pool
+	pool, err := vm.Create(cfg, true)
+	if err != nil {
+		return fmt.Errorf("Failed to create a VM Pool : %v", err)
+	}
+	// Create a VM instance (We need only one)
+	inst, err := pool.Create(0)
+	if err != nil {
+		return fmt.Errorf("Failed to create the VM Instance : %v", err)
+	}
+	defer inst.Close()
+	// Copy the kernel into the disk image and replace it
+	kernel, err := inst.Copy(filepath.Join(outputDir,"netbsd"))
+	if err != nil {
+		return fmt.Errorf("Error Copying the kernel %v: %v", kernel, err)
+	}
+	outc, errc, err := inst.Run(time.Minute, nil, "sync")
```

Comment:
inst.Run does not wait for the command to finish. It just starts it and returns channels that stream command output and errors. This is the method we use to start the fuzzer in the VMs.
We also need to run vm.MonitorExecution and wait till it returns nil.

Reply:
Hmm.. What about not running sync and just running poweroff? I think sync is not necessarily required. I had added it there because I thought the image not getting copied was due to some other issue.

Reply:
In that case do we still require vm.MonitorExecution?

Reply:
If we don't use inst.Run, then we don't need vm.MonitorExecution (there will be simply nothing to pass to the function).
But will the data for the new file be persisted without sync? I not sure what will guarantee that.

Reply:
I was having some difficulty with the vm.MonitorExecution part. Maybe because it was power off. So I am just sleeping for a while until the poweroff happens.
It is a pretty primitive solution. Should I try it with vm.MonitorExecution itself?